afpacket/afpacket.go

```diff
@@ -293,8 +295,8 @@ retry:
 
 // Stats returns statistics on the packets the TPacket has seen so far.
 func (h *TPacket) Stats() (Stats, error) {
-	h.mu.Lock()
-	defer h.mu.Unlock()
+	h.statsMu.Lock()
```

Comment:
You still want mu here instead of statsMu, I think, due to line 290 and 427 increment of the # of packets/polls.  Or you could consider replacing those increments with sync/atomic.AddInt64.